Training/Training/main.c

```diff
@@ -0,0 +1,185 @@
+#include<stdio.h>
+#include<malloc.h>
+#include<math.h>
+
+int mCount;
+
+//Function for counting the number of digits
+int count (int dec, int div) {
+   int c = 0;
+   if (dec == 0)
+      return 8;
+   while (dec > 0) {
+      dec /= div;
+      c++;
+   }
+   return c;
+}
+
+int* DecToBin (int dec) {
+   mCount = 0;
+   int dec2 = dec;
+   if (dec < 0)
+      dec *= -1;
+   int dec1 = dec;
+   //Counting the number of digits in binary representation
+   mCount = count (dec, 2);
+   // Representing binary in multiples of 8-bits accordingly
+   if (mCount % 8 != 0)
+      mCount += (8 - (mCount % 8));
+   int* binRev = NULL;
+   do {
+      binRev = (int*)malloc (mCount * sizeof (int));
+   } while (binRev == NULL);
+   //Storing the remainders into binRev
+   for (int i = 0; i < mCount; i++) {
+      binRev[i] = dec1 % 2;
+      dec1 /= 2;
+   }
+   if (dec2 >= 0) {
+      int* bin;
+      do {
+         bin = (int*)malloc (mCount * sizeof (int));
+      } while (bin == NULL);
+      // Reverse storing the stored remainders
+      for (int j = 0; j < mCount; j++)
+         bin[j] = binRev[mCount - j - 1];
+      return bin;
+   } else { //Performing two's compliment
+      int* binTwosComp = NULL;
+      do {
+         binTwosComp = (int*)malloc (mCount * sizeof (int));
+      } while (binTwosComp == NULL);
+      //Complimenting each element of the array
+      for (int k = 0; k < mCount; k++) {
+         if (binRev[mCount - k - 1] == 0)
+            binTwosComp[k] = 1;
+         else
+            binTwosComp[k] = 0;
+      }
+      free (binRev);
+      //Adding 1 to the the binary number (stored as individual digits)
+      for (int k = mCount - 1; k >= 0; k--) {
+         if (binTwosComp[k] == 0) {
+            binTwosComp[k] = 1;
+            break;
+         } else
+            binTwosComp[k] = 0;
+      }
+      return binTwosComp;
+   }
+}
+
+char* DecToHex (int dec) {
+   mCount = 0;
+   int dec2 = dec;
+   if (dec < 0)
+      dec *= -1;
+   int dec1 = dec;

```

Comment:
The names for variables dec1 and dec2 can be more meaningful. For example, int dec, dec2, dec1 can be named as n, nTemp, negN respectively which will explain its purpose by just reading.